### DISCUSSION TITLE
How to filter points from first meter from the surface of the upper canopy?

### DISCUSSION THREAD

**Discussion by elaliberte** [r-lidar/lidR#626] [Q&A]
I have a dense colorized point cloud derived from drone imagery and photogrammetry and I would like to only keep points that fall within the first (uppermost) meter of the upper canopy. In other words, only points that would fall between the DSM height and DSM height - 1 m. The reason I want to do this is to extract greenness of tree crowns throughout a growing season (I have repeat imagery/point clouds) but at the start of the season deciduous trees are leafless but the understory is sometimes green and measuring greenness from the 2D RGB orthomosaic would strongly overestimate greennness when trees are leafless or partially leafless. Considering only the top of the point clouds could solve that issue.
In other words I would like to slice only the first meter of foliage and discard the rest.
Is this possible using lidR? I don't have returns because this isn't lidar data, and I can't just filter points between a certain elevation range because trees differ in height plus there is undulating topography too.
Thanks.

**Comment by Jean-Romain** [comment]
Compute a DSM with `rasterize_canopy()`, merge the data with `merge_spatial(las, dsm, "Zdsm")`, filter with `filter_poi(las, Z > Zdsm - 1)`

**Reply by elaliberte** [reply]
Wow, thank you so much - it worked!!
